nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LookupRecord.java

```diff
@@ -548,8 +577,15 @@ public Set<Relationship> lookup(final Record record, final ProcessContext contex
 
             final FlowFile flowFile = lookupContext.getOriginalFlowFile();
             final Optional<?> lookupValueOption;
+            final Optional<?> lookupValueCacheOption;
+
             try {
-                lookupValueOption = lookupService.lookup(lookupCoordinates, flowFile.getAttributes());
+                lookupValueCacheOption = (Optional<?>) cache.get(lookupCoordinates, k -> null);
```

Comment:
fyi, you can use `cache.getIfPresent(key)` to lookup or return null. Of course if possible it is preferred to compute within the cache to avoid a cache stampede.